### DISCUSSION TITLE
Using BeforeInsert or BeforeUpdate hooks to change a column value. How to?

### DISCUSSION THREAD

**Discussion by frederikhors** [uptrace/bun#244] [Q&A]
I'm trying to use the `BeforeInsertHook` like in the example below:

```go
var _ bun.BeforeInsertHook = (*Model)(nil)

func (*Model) BeforeInsert(ctx context.Context, query *bun.InsertQuery) error {
  // get customCode here...

  // update a column of the model here with customCode, ex:

  // m.CustomCodeColumn = customCode
}
```

But I cannot find a way.

What should I use?

UPDATE: I think we should update the docs too for this case.

**Comment by vmihailenco** [comment]
There is an idea to add `BeforeAppendModel` hook that is called before a model is append to a query - https://github.com/uptrace/bun/issues/194

It should be possible with existing hooks too, but it is not trivial - see https://github.com/uptrace/bun/blob/master/example/model-hooks/main.go#L68-L97

**Reply by frederikhors** [reply]
I'm using right now this:

```go
var _ bun.BeforeInsertHook = (*Menu)(nil)

func (*Menu) BeforeInsert(ctx context.Context, query *bun.InsertQuery) error {
	tenantID := auth.GetTenantID(ctx)

	query.Value("tenant_id", strconv.Itoa(tenantID))

	return nil
}
```

But I don't like the usage of `strconv` and I don't know if the `.Value()` method is the way to go.

Plus, I don't know if this is a problem with `[]Menu` instead of a single `Menu`.

**Reply by frederikhors** [reply]
@vmihailenco Can you help me closing this issue?

**Reply by vmihailenco** [reply]
Please try https://bun.uptrace.dev/guide/hooks.html#beforeappendmodel . It is experimental and you will need to use the latest master branch to try it.

**Reply by frederikhors** [reply]
Wow. You were fast!

I think I made it works with `query.Value("tenant_id", strconv.Itoa(tenantID))`.

It works both with multiple or single insert. I'm happy with it!

Is there anything I haven't considered?

**Reply by vmihailenco** [reply]
It should work I guess :)
